app/controllers/snippets_controller.rb

```diff
@@ -49,21 +49,39 @@ def update
     end
   end
 
+  def discuss
+    @snippet = Snippet.find(params[:id])
+    return redirect_to @snippet.slack_url if @snippet.slack_url
+
+    text = "`SOLUTION` Hey <@#{@snippet.user.slack_id}>, some people want to discuss your :#{@snippet.language}-hd: #{solution_markdown} on puzzle #{@snippet.day} part #{@snippet.challenge}"
+    message = client.chat_postMessage(channel: ENV.fetch("SLACK_CHANNEL", "#aoc-dev"), text:)
+    slack_thread = client.chat_getPermalink(channel: message["channel"], message_ts: message["message"]["ts"])
+    @snippet.update(slack_url: slack_thread[:permalink])
+
+    redirect_to @snippet.slack_url
+  end
```

Comment:
On pourrait extraire ça dans sa propre classe je me dis.
Style:
```suggestion
    @snippet = Snippet.find(params[:id])
    thread = Snippets::OpenDiscussionThread.call(@snippet)
    
    return redirect_to @snippet.slack_url
  end
```